lib/SILGen/SILGenProlog.cpp

```diff
@@ -423,6 +424,26 @@ void SILGenFunction::emitProlog(AnyFunctionRef TheClosure,
                                 Type resultType, bool throws) {
   uint16_t ArgNo = emitProlog(paramList, selfParam, resultType,
                               TheClosure.getAsDeclContext(), throws);
+  
+  // Emit an unreachable instruction if a parameter type is
+  // uninhabited
+  if (paramList) {
+    for (auto *param : *paramList) {
+      if (param->getType()->isStructurallyUninhabited()) {
+        SILLocation unreachableLoc(param);
+        unreachableLoc.markAsPrologue();
+        B.createUnreachable(unreachableLoc);
+        
+        if (auto body = TheClosure.getBody()) {
+          if (!body->getElements().empty()) {
+            auto &ctx = SGM.M.getASTContext();
+            ctx.Diags.diagnose(body->getStartLoc(), diag::unreachable_code_uninhabited_param_note, param->getName());
```

Comment:
Unfortunately, this isn't allowed: notes must always be emitted following the actual warning or error they should be attached to. This'll have to go at the point of diagnosing the unreachable code.

Reply:
Oh, do you suggest I put the note there, or just change the note here to a warning?

Reply:
I agree, I'd prefer adding the note to the eventual warning as well, if it's not too much trouble.

Reply:
Done @jrose-apple @jckarter :)